src/dialects/db2/query-generator.js

```diff
@@ -751,59 +763,114 @@ export class Db2QueryGenerator extends AbstractQueryGenerator {
+   * @param {string|object|Array} table
+   *        The name of the table as a tring
+   *        The tableName and/or schema in an object
+   *        An array of objets
+   *
+   * @param {string} _database
+   *        The name of the database
+   *
+   * @param {object} [options]
+   * @param {string} [options.schema]
+   *        The name of the schema
+   *
+   * @returns {string} The generated sql query.
    */
-  getForeignKeysQuery(table, schemaName) {
-    const tableName = table.tableName || table;
-    schemaName = table.schema || schemaName;
-    let sql = '';
-    if (tableName) {
-      sql = ` AND R.TABNAME = ${wrapSingleQuote(tableName)}`;
-    }
+  getForeignKeysQuery(table, _database, options) {
+    let tableName = table.tableName || table;
+    let schemaName = table?.schema || options?.schema || '';
 
-    if (schemaName) {
-      sql += ` AND R.TABSCHEMA = ${wrapSingleQuote(schemaName)}`;
-    }
+    const tableDetails = Array.isArray(table) ? table : [{ tableName, schema: schemaName }];
+
+    /*
+      Example expected translation:
+        FROM:  [ {tableName: 'foo'}, {tableName: 'bar', schema: 'private'}, {}]
+        TO:    "( ref.tabname = 'foo' ) OR ( ref.tabname = 'bar' AND ref.tabschema = 'private' ) "
 
-    return this._getForeignKeysQuerySQL(sql);
+      Syntax Help:
+        `filter(v => !!v)` is used to discard falsy values ( or keep truthy)
+    */
+    let whereConditions = tableDetails.map(tableAndOrSchema => {
+      tableName = tableAndOrSchema.tableName || '';
+      schemaName = tableAndOrSchema?.schema || options?.schema || '';
+
+      const expressions = [];
+
+      if (tableName) {
+        expressions.push(`fktable_name = ${this.escape(tableName)}`);
+      }
+
+      if (schemaName) {
+        expressions.push(`fktable_schem = ${this.escape(schemaName)}`);
+      }
+
+      return expressions.filter(v => !!v).join(' AND ');
+    });
+
+    // filter out undefineds & blank strings and wrap in parentheses
+    whereConditions = whereConditions.filter(v => !!v).map(expr => `( ${expr} )`).join(' OR ');
+
+    return this._getForeignKeysQuerySQL(whereConditions);
   }
 
   getForeignKeyQuery(table, columnName) {
     const tableName = table.tableName || table;
     const schemaName = table.schema;
-    let sql = '';
+
+    const whereConditions = [];
     if (tableName) {
-      sql = ` AND R.TABNAME = ${wrapSingleQuote(tableName)}`;
+      whereConditions.push(` AND R.TABNAME = ${wrapSingleQuote(tableName)}`);
     }
 
     if (schemaName) {
-      sql += ` AND R.TABSCHEMA = ${wrapSingleQuote(schemaName)}`;
+      whereConditions.push(` AND R.TABSCHEMA = ${wrapSingleQuote(schemaName)}`);
     }
 
     if (columnName) {
-      sql += ` AND C.COLNAME = ${wrapSingleQuote(columnName)}`;
+      whereConditions.push(` AND C.COLNAME = ${wrapSingleQuote(columnName)}`);
     }
 
-    return this._getForeignKeysQuerySQL(sql);
+    return this._getForeignKeysQuerySQL(whereConditions.join(''));
```

Comment:
I don't think this change is necessary, it does exactly the same thing as before and this PR is big enough already

Reply:
It's not _necessary_.  It's a general-practice performance tweak when working with strings.

Reply:
It's not more performant though. String concatenation is not faster than populating an array and joining it :)